examples/src/fungible-token.js

```diff
@@ -1,87 +1,136 @@
+    assert(!this.accounts.containsKey(accountId), "Account is already registered");
+    this.accounts.set(accountId, "0");
+    this.accountRegistrants.set(accountId, registrantAccountId);
+    this.accountDeposits.set(accountId, amountStr);
   }
 
-  internalWithdraw({ accountId, amount }) {
-    let balance = this.accounts.get(accountId, { defaultValue: 0n });
-    let newBalance = balance - BigInt(amount);
-    assert(newBalance >= 0n, "The account doesn't have enough balance");
-    this.accounts.set(accountId, newBalance);
-    let newSupply = this.totalSupply - BigInt(amount);
-    assert(newSupply >= 0n, "Total supply overflow");
-    this.totalSupply = newSupply;
+  internalSendNEAR(receivingAccountId, amountBigInt) {
+    assert(amountBigInt > BigInt("0"), "The amount should be a positive number");
+    assert(receivingAccountId != near.currentAccountId(), "Can't transfer to the contract itself");
+    assert(amountBigInt < near.accountBalance(), `Not enough balance ${near.accountBalance()} to cover transfer of ${amountBigInt} yoctoNEAR`);
+    const transferPromiseId = near.promiseBatchCreate(receivingAccountId);
+    near.promiseBatchActionTransfer(transferPromiseId, amountBigInt);
+    near.promiseReturn(transferPromiseId);
   }
 
-  internalTransfer({ senderId, receiverId, amount, memo: _ }) {
+  internalGetBalance(accountId) {
+    assert(this.accounts.containsKey(accountId), `Account ${accountId} is not registered`);
+    return this.accounts.get(accountId);
+  }
+
+  internalDeposit(accountId, amount) {
+    let balance = this.internalGetBalance(accountId);
+    let newBalance = BigInt(balance) + BigInt(amount);
+    this.accounts.set(accountId, newBalance.toString());
+    let newSupply = BigInt(this.totalSupply) + BigInt(amount);
+    this.totalSupply = newSupply.toString();
+  }
+
+  internalWithdraw(accountId, amount) {
+    let balance = this.internalGetBalance(accountId);
+    let newBalance = BigInt(balance) - BigInt(amount);
+    assert(newBalance >= BigInt(0), "The account doesn't have enough balance");
+    this.accounts.set(accountId, newBalance.toString());
+    let newSupply = BigInt(this.totalSupply) - BigInt(amount);
+    assert(newSupply >= BigInt(0), "Total supply overflow");
+    this.totalSupply = newSupply.toString();
+  }
+
+  internalTransfer(senderId, receiverId, amount, memo = null) {
     assert(senderId != receiverId, "Sender and receiver should be different");
-    let amountInt = BigInt(amount);
-    assert(amountInt > 0n, "The amount should be a positive number");
-    this.internalWithdraw({ accountId: senderId, amount });
-    this.internalDeposit({ accountId: receiverId, amount });
+    assert(BigInt(amount) > BigInt(0), "The amount should be a positive number");
+    this.internalWithdraw(senderId, amount);
+    this.internalDeposit(receiverId, amount);
+  }
+
+  @call({ payableFunction: true })
+  storage_deposit({ account_id }) {
```

Comment:
Hm... why are we using JS for this example? Types will be beneficial here.

Reply:
Good point actually. Will adapt to TS and request re-review